sherpa/models/model.py

```diff
@@ -1357,13 +1357,25 @@ def guess(self, dep, *args, **kwargs):
         pass
 
 
-def _make_unop(op: Callable, opstr: str) -> Callable:
-    def func(self):
-        return UnaryOpModel(self, op, opstr)
+# It's hard to come up with a sensible typing rule for the operator
+# in _make_unop/binop so we just use Callable.
+#
+def _make_unop(op: Callable,
+               opstr: str,
+               strformat: Optional[str] = None) -> Callable:
```

Comment:
```suggestion
               strformat: str | None = None) -> Callable:
```

[since we have dropped Python 3.9 suport]

Reply:
or support...